lib/everest.cpp

```diff
@@ -486,6 +486,13 @@ void Everest::subscribe_var(const Requirement& req, const std::string& var_name,
         EVLOG_verbose << fmt::format(
             "Incoming {}->{}", this->config.printable_identifier(requirement_module_id, requirement_impl_id), var_name);
 
+        // A race condition where some other module might received its on-ready
+        // and this module not (and calling us).
+        while (!ready_received) { // In C++20 we might mark it as [[unlikely]]
```

Comment:
better move this into a function that can be reused in the other cases its needed